lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -352,15 +352,43 @@ class DetailsRenderer {
    */
   renderNode(item) {
     const element = this._dom.createElement('span', 'lh-node');
-    if (item.snippet) {
-      element.textContent = item.snippet;
-    }
     if (item.selector) {
       element.title = item.selector;
     }
     if (item.path) element.setAttribute('data-path', item.path);
     if (item.selector) element.setAttribute('data-selector', item.selector);
     if (item.snippet) element.setAttribute('data-snippet', item.snippet);
+    element.innerHTML = `<div style="color: black;font-family: var(--text-font-family);margin-bottom: 10px;">${item.textContent}</div>`;
+    const snippetEl = document.createElement('div');
+    snippetEl.style.fontSize = '12px';
+    snippetEl.innerText = item.snippet;
+    element.appendChild(snippetEl);
+
+    const fullpageScreenshotUrl = __LIGHTHOUSE_JSON__.audits['full-page-screenshot'].details.data;
+
+    if (item.boundingRect) {
+      /** @type {Element} */
+      let elementScreenshot;
+      element.addEventListener('mouseenter', () => {
```

Comment:
not sure what our stance is on JS in report now

@paulirish 🤓 🔫 , CSS-only impl? ;)

Reply:
I can't find the comment, but I think we decided that's ok on the snippet renderer PR.

Not sure how feasible this would be to do without JS, browsers might not like having a hundred 500kb base64 images on one page.

Reply:
Oh that sounds right. I was remembering some I/O convo about going back and forth to make things easier on PSI, but we did already break that barrier.